crates/autopilot/src/limit_orders/quoter.rs

```diff
@@ -58,10 +58,13 @@ impl LimitOrderQuoter {
 
     /// Returns whether it is likely that there is no more work.
     async fn update(&self) -> Result<bool> {
-        let order_specs = self
+        let orders = self
             .database
-            .order_specs_with_outdated_fees(self.limit_order_age, self.parallelism)
+            .open_limit_orders(self.limit_order_age)
             .await?;
+        let orders = self.orders_with_sufficient_balance(orders).await;
+        let order_specs: Vec<_> = orders.into_iter().map(order_spec_from).unique().collect();
```

Comment:
Not a suggestion, but something that I like (subjective and esthetic) in projects that already import `itertools` is to
```suggestion
        let order_specs = orders.into_iter().map(order_spec_from).unique().collect_vec();
```